### ISSUE TITLE
Male voice

### ISSUE THREAD

**Issue by loretoparisi** [ming024/FastSpeech2#11]
First thank you, I have solved the issue opened thanks to you support. In my understanding both Melgan (that I have tried) and Waveglow (not run yet) have a female voice. To have a male voice is it necessary to train from scratch the model? Or add support to a specific vocoder?

Thank you.

**Comment by ming024** [comment]
@loretoparisi  I think that (Universal Vocoding)[https://github.com/bshall/UniversalVocoding] meets your requirement. I have a plan to support multi-speaker TTS, but it is not my top priority now. You are welcomed to fork this repository for a male-TTS  implementation. I think both the f0_min for PyWorld Vocoder and some parameter related to preprocessing have to be modified for male speakers.

**Comment by shoegazerstella** [comment]
HI @ming024 
So your suggestion would be to integrate Universal Vocoder into your implementation of FastSpeech2 ?

**Comment by shoegazerstella** [comment]
Hi @ming024,
I am trying what you suggested, to use Universal Vocoder for synthesizing the output of FastSpeech2.

The output of FS2 has size:
`torch.Size([1, 80, x])`

while UV wants something like this as input:
`torch.Size([1, x, 80])`

I tryed by swapping the axis but of course didn't work, what I got was just noise or silence.
So this confirms what you suggested, that maybe it is worth retraining UV with the same parameters as FS2.

Another thing I tried was playing with the UV params, using the mel_spec computed from the generated wav file by FS2. I got some interesting changes in pitch, but nothing I can really use for my purpose of changing the speaker.

If you have any other advice I could use please let me know, thanks a lot!

**Comment by ming024** [comment]
@shoegazerstella Actually I haven't tried Universal Vocoding before so I am not sure where the error come from. 

I think the decoder of FastSpeech is similar to the decoder of a voice conversion model. Some VC models use vector quantization or other tricks to learn a discrete embedding space, and find out that phonetic information is contained therein. Maybe there is some way to combine the training of both tasks, no matter jointly training or pretraining, etc.

**Comment by carankt** [comment]
@ming024 Any idea what specific changes one needs to make for Male voice cloning?

**Comment by joseluismoreira** [comment]
@carankt  any updates about male voice or how to do it, please?

**Comment by ming024** [comment]
Guys multi-speaker synthesis is supported now.

**Comment by ming024** [comment]
closed #11
